broker-core/src/main/java/io/zeebe/broker/logstreams/processor/TypedStreamProcessor.java

```diff
@@ -159,16 +162,38 @@ public void wrap(
 
     @Override
     public void processEvent() {
-      writer.reset();
-      responseWriter.reset();
+      try {
+
+        writer.reset();
+        responseWriter.reset();
 
-      this.writer.configureSourceContext(streamProcessorId, position);
+        this.writer.configureSourceContext(streamProcessorId, position);
 
-      // default side effect is responses; can be changed by processor
-      sideEffectProducer = responseWriter;
+        // default side effect is responses; can be changed by processor
+        sideEffectProducer = responseWriter;
 
-      eventProcessor.processRecord(
-          position, event, responseWriter, writer, this::setSideEffectProducer);
+        eventProcessor.processRecord(
+            position, event, responseWriter, writer, this::setSideEffectProducer);
+      } catch (Exception exception) {
+        final String errorMessage =
+            String.format(PROCESSING_ERROR_MESSAGE, event, exception.getMessage());
+        LOG.error(errorMessage, exception);
+
+        if (event.metadata.getRecordType() == RecordType.COMMAND) {
+          sendCommandRejectionOnException(errorMessage);
+        } else if (event.getMetadata().getRecordType() == RecordType.EVENT) {
+          // TODO(#2028) clean up state
+        }
+
+        // re-throw such that stream process controller skips this record
+        throw exception;
```

Comment:
I would like that we have a mechanism for `StreamProcessor` implementations to specify what to do on error as opposed to catching and rethrowing the exception, but we can implement that later if you think it's more refactoring/outside the scope here.

Reply:
Yeah would be nice, but I would like to postpone this for now, since we have no real need and is out of scope of this issue. Feel free to create an follow up issue.